### DISCUSSION TITLE
Register size

### DISCUSSION THREAD

**Discussion by gvalderramos** [skypjack/entt#971] [Q&A]
Hi guys,

I'm testing this lib that requires a lot of entities. However the reason, eventually I'm getting an error saying that there are no more entities available....is there a way to set it bigger?

// I'm trying to create a vertex entity with:
auto vertex_entity = m_registry.create();

// I'm getting this assertion error:
entt.hpp:34758: auto entt::basic_registry<Entity, <template-parameter-1-2> >::generate_identifier(std::size_t) [with Entity = entt::entity; Allocator = std::allocator<entt::entity>; std::size_t = long unsigned int]: Assertion `pos < entity_traits::to_entity(null)' failed.

Thank you, guys.

**Comment by skypjack** [comment]
Yeah, sure. A 32b identifier uses 20b for the entities and 12b for the version. You've a few possibilities here:

* Change the split above and use more bits for the entity. You can do that by specializing the `entt_traits` class.

* Use a 64b identifier, either globally or as a custom one. You can define your own entity type as in:
    ```cpp
    enum class my_entity_type: std::uint64_t {};
    ```
    Or as a class that makes the underlying type available to the library:
    ```cpp
    struct my_entity_type {
        using entity_type = std::uint64_t;
        // ...
    };
    ```
    The second option let you add more functionalities to the entity type too if you like/need to.
    By default, 64b identifiers use 32b for the entitities and it's likely more than enough for your needs.
    Then, you can use your entity type to instantiate the registry: `entt:.basic_registry<my_entity_type>` and you're done.

* Otherwise, you can override the `ENTT_ID_TYPE` variable and set it to `std:.uint64_t` (although this also affects a few other things around).

Out of curiosity, what are you working on for which 1M entities isn't enough? That's really a huge number after all.

**Reply by gvalderramos** [reply]
Hey @skypjack, thank you for replaying me.

When I changed the ENTT_ID_TYPE to std::uint64_t kind of worked because I didn't get this error anymore but the process to create the entities took forever...
My use case is I have a crowd system library that will run hundreds of instances of actors and I'm trying to use entt lib to represent the actor geometry data, like positions, normals of each vertex, and face of these actors' instances. The legacy code uses only container classes to store this data, I'm just trying to see if Entt will represent this data better and faster.
But I don't know if I'm using the right architecture, maybe I'm creating too much data with the same register. I'm using one register to represent this scene that has all these actors. The problem is the geometry can have a huge data...so if I create one entity by polygon, for instance, to store the normal position, it just crashes.

**Reply by skypjack** [reply]
It's hard to say what's happening without looking at the code.
From what you describe, you're probably running out of memory?
